### PR TITLE
AdapterDelegatesManager: add method to get a delegate by view type

### PR DESCRIPTION
**Pull request by liangricha**

### CODE REVIEW

**Comment by sockeqwe** [comment]
What would be the use case for this?

**Comment by liangricha** [comment]
Allows users of this library to extend managers and/or delegates and include additional RV callbacks like `onViewRecycled(...)`, `onFailedToRecycle(...)`, etc

File: library/src/main/java/com/hannesdorfmann/adapterdelegates2/AdapterDelegatesManager.java [hunk 1]
```diff
@@ -310,4 +302,27 @@ public int getViewType(@NonNull AdapterDelegate<T> delegate) {
...
+  /**
+   * Get the {@link AdapterDelegate} associated with the given view type integer
+   * @param viewType The view type integer we want to retrieve the associated
+   *                 delegate for.
+   * @return The {@link AdapterDelegate} associated with the view type param if it exists,
+   * the fallback delegate otherwise if it is set.
+   * @throws NullPointerException if no delegates are associated with this view type
+   * and if no fallback delegate is set.
+   */
+  @NonNull public AdapterDelegate<T> delegateForViewType(int viewType) {
```
Comment (sockeqwe) [inline library/src/main/java/com/hannesdorfmann/adapterdelegates2/AdapterDelegatesManager.java]: Please rename this method to `getDelegateForViewType(int)`

**Comment by liangricha** [comment]
@sockeqwe renamed that method!

**Comment by sockeqwe** [comment]
Thanks!

**Pull request merged**